file_browser/src/components/FolderView.tsx

```diff
@@ -0,0 +1,92 @@
+import { SystemItem } from "../classes/SystemItem";
+import { Folder } from "../classes/Folder";
+import { observer } from "mobx-react-lite";
+import { useState } from "react";
+import fileSystemStore from "../stores/FileSystemStore";
+import { FileView } from "./FileView";
+import { FileClass } from "../classes/File";
+
+// icons
+import openedIcon from "../assets/openedFolder.png";
+import whiteOpenedIcon from "../assets/openedFolderWhite.png";
+import closedIcon from "../assets/closedFolder.png";
+import whiteClosedIcon from "../assets/closedFolderWhite.png";
+import leftArrow from "../assets/chevron.png";
+import whiteLeftArrow from "../assets/whiteLeftArrow.png";
+import downArrow from "../assets/down-arrow.png";
+import whiteDownArrow from "../assets/whiteDownArrow.png";
+
+const FolderView = observer(({ folder }: { folder: Folder }) => {
+  const [isFolderOpened, setIsFolderOpened] = useState(false);
+
+  const folderIcon =
+    isFolderOpened && folder.files.size !== 0
+      ? fileSystemStore.selectedId === folder.id
+        ? whiteOpenedIcon
+        : openedIcon
+      : fileSystemStore.selectedId === folder.id
+      ? whiteClosedIcon
+      : closedIcon;
+
+  const arrowIcon = isFolderOpened
+    ? fileSystemStore.selectedId === folder.id
+      ? whiteDownArrow
+      : downArrow
+    : fileSystemStore.selectedId === folder.id
+    ? whiteLeftArrow
+    : leftArrow;
```

Comment:
Ternary operator is only good for 2 short options. That's all.
Please change this to memos with `if/else` chains.